testslide/matchers.py

```diff
@@ -191,31 +191,55 @@ def __init__(self) -> None:
 
 class NotThisInt(_IntComparison):
     def __init__(self, ne: int) -> None:
+        if not isinstance(ne, int):
+            raise ValueError(
+                f"NotThisInt(...) expects an 'int' as argument while '{type(ne).__name__}' was provided"
+            )
         super().__init__(ne=ne)
 
 
 class IntBetween(_IntComparison):
     def __init__(self, lower: int, upper: int) -> None:
+        if not isinstance(lower, int) or not isinstance(upper, int):
```

Comment:
I would personally allow float values as well.
I mean what if I'm looking for any integer between 2.8 and 33.9 ?

Think about a case like
```python
def test_dummy(self) -> None:
    lower = some_calculation()  # This returns a number, maybe an integer value but probably stored in float
    upper = lower * 2
    IntBetween(lower, upper)  # This would fail, but from a conceptual point of view it might be valid
```

This comment is valid for all the Int* matchers